languages/python/oso/polar/polar.py

```diff
@@ -151,6 +152,7 @@ def load_str(self, string, filename=None):
             if query is None:  # Load is done
                 break
             else:
+                query.set_message_enricher(self.host.enrich_message)
```

Comment:
Same here, use the copied host.

Reply:
This is really error prone & these would be very hard errors to detect through testing. I think `query.Query` (not `ffi.Query`) can call `ffi.Query.set_message_enricher` in its initializer to ensure that the host classes on the two objects match.

Reply:
Ah yeah that makes a lot more sense. I just fixed that and its way clearer I think